File: server/Models/Location.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace RedHouse_Server.Models
{
    public class Location
    {
        [Key]
        public int LocationId { get; set; }
        public string ? StreetAddress { get; set; }
        public string ? City { get; set; }
        public string ? Region { get; set; }
        public string?  PostalCode { get; set; }
        public string ? Country { get; set; }
        public double ? Latitude { get; set; }
        public double ? Longitude { get; set; }
    }
}

File: mobile/lib/model/location.dart
// ignore_for_file: non_constant_identifier_names
import 'package:json_annotation/json_annotation.dart';

part "location.g.dart";

@JsonSerializable()
class Location {
  int id;
  String streetAddress;
  String city;
  String region;
  String postalCode;
  String country;
  double latitude;
  double longitude;

  Location({
    required this.id,
    required this.streetAddress,
    required this.city,
    required this.region,
    required this.postalCode,
    required this.country,
    required this.latitude,
    required this.longitude,
  });

  factory Location.fromJson(Map<String, dynamic> json) =>
      _$LocationFromJson(json);
  Map<String, dynamic> toJson() => _$LocationToJson(this);
}
